lib/rspec/mocks/verifying_proxy.rb

```diff
@@ -4,6 +4,21 @@
 module RSpec
   module Mocks
     # @private
+    class CallbackInvocationStrategy
+      def call(doubled_module)
+        RSpec::Mocks.configuration.verifying_double_callbacks.each do |block|
+          block.call doubled_module
+        end
+      end
+    end
+
+    # @private
+    class NoCallbackInvocationStrategy
+      def call(_doubled_module)
+      end
+    end
```

Comment:
Personally, using the strategy pattern to conditionally do something feels like overkill.  The strategy pattern has its uses but here it feels to me like it's being used just to avoid an `if` conditional, and IMO the `if` conditional would actually be simpler and better.  Anytime you're adding a conditional I think it's worth considering what a polymorphic solution would look like...but not always worth choosing it over the conditional.

Anyhow, if you do decide to keep using the strategy pattern here, I'd advocate for making these singleton modules instead of classes you instantiate since they are stateless.  Fewer object instantiates means less garbage for the GC to cleanup :).